File: poc/nutricalc/energy.py
"""Besoin énergétique journalier.

Métabolisme de base (BMR) par l'équation de Mifflin-St Jeor (1990), puis
besoin énergétique total (TDEE) = BMR × facteur d'activité.
"""

from __future__ import annotations

from .profile import Profile, Sex


def bmr_mifflin_st_jeor(profile: Profile) -> float:
    """Métabolisme de base en kcal/j (Mifflin-St Jeor)."""
    base = 10 * profile.weight_kg + 6.25 * profile.height_cm - 5 * profile.age_years
    return base + (5 if profile.sex is Sex.MALE else -161)


def tdee(profile: Profile) -> float:
    """Besoin énergétique total en kcal/j (BMR × niveau d'activité)."""
    return bmr_mifflin_st_jeor(profile) * float(profile.activity.value)

File: src/calc/energy.ts
// Besoin énergétique journalier : BMR Mifflin-St Jeor, TDEE, puis application de
// l'objectif calorique du profil nutritionnel.
import {
  ACTIVITY_LEVELS,
  NUTRITION_PROFILES,
  type CalorieGoal,
  type Profile,
} from "./profile";

export function bmrMifflinStJeor(p: Profile): number {
  const base = 10 * p.weightKg + 6.25 * p.heightCm - 5 * p.ageYears;
  return base + (p.sex === "male" ? 5 : -161);
}

// Dépense énergétique totale : le BMR est calculé sur le poids réel (la dépense
// dépend de la masse effectivement portée).
export function tdee(p: Profile): number {
  return bmrMifflinStJeor(p) * ACTIVITY_LEVELS[p.activity];
}

export interface EnergyTarget {
  tdeeKcal: number;
  goal: CalorieGoal;
  adjustmentKcal: number; // ajustement cible appliqué au TDEE
  energyKcal: number; // TDEE + ajustement cible
  energyMinKcal: number; // borne basse de la fourchette
  energyMaxKcal: number; // borne haute de la fourchette
}

// Applique l'objectif calorique : calories = TDEE + ajustement (négatif en
// déficit). La fourchette reflète l'intervalle d'ajustement du profil.
export function energyTarget(p: Profile, tdeeKcal = tdee(p)): EnergyTarget {
  const np = NUTRITION_PROFILES[p.goal];
  const adj = np.adjustmentKcal;
  return {
    tdeeKcal,
    goal: np.goal,
    adjustmentKcal: adj.target,
    energyKcal: tdeeKcal + adj.target,
    energyMinKcal: tdeeKcal + adj.min,
    energyMaxKcal: tdeeKcal + adj.max,
  };
}
